packages/core-transaction-pool/lib/interface.js

```diff
@@ -209,34 +219,51 @@ module.exports = class TransactionPoolInterface {
         ? this.walletManager.findByAddress(transaction.recipientId)
         : false
 
-      // If sender or recipient wallet in pool we try to apply transaction
-      const exists = this.transactionExists(transaction.id)
-      if (!exists && (senderWallet || recipientWallet)) {
-        try {
-          this.walletManager.applyPoolTransaction(transaction)
-        } catch (error) {
-          logger.error(`AcceptChainedBlock in pool: ${error}`)
-          // Purge sender
-          this.purgeByPublicKey(senderPublicKey)
-        }
-      } else {
-        this.removeTransaction(transaction)
+      if (recipientWallet) {
+        recipientWallet.applyTransactionToRecipient(transaction)
       }
 
-      if (
-        senderWallet &&
-        senderWallet.balance === 0 &&
-        this.getSenderSize(senderPublicKey) === 0
-      ) {
-        this.walletManager.deleteWallet(senderPublicKey)
+      if (exists) {
+        this.removeTransaction(transaction)
+      } else if (senderWallet) {
+        const errors = []
+        if (senderWallet.canApply(transaction, errors)) {
```

Comment:
I think there is on requirement for transactions in a block to be in any specific order. If this is the case, then we should not be upset here if we cannot apply a given tx. Maybe another one in the block would give that sender enough funds.

Reply:
Not true. Block is already applied and forged before we enter this function and the definitive and true state of the chain. Here we apply that state if needed, and if not successful the senders must go out of pool.

Order is important for other things, vote/unvote etc, but not related to this.